### PR TITLE
[AIRFLOW-6088] pass DAG processing runtime as duration to stats

### PR DESCRIPTION
**Pull request by hoesler**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title.
  - https://issues.apache.org/jira/browse/AIRFLOW-6088

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:
runtime var is now a duration object, not a float

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

**Review by XD-DENG — APPROVED** [review]
It's a nice catch.
I cross-checked the code & documentation, and LGTM.
Thanks for your contribution!

**Comment by XD-DENG** [comment]
@hoesler Thanks again for the PR! LGTM and I would be more than happy to merge it to master.

Before I merge, may we request a favor from you: may you check if there is similar/same issue for other `statsd`-related code? If yes, we can address them together in this PR.

I will also do the same check from my side. After that we are good to go.

**Comment by XD-DENG** [comment]
Hi @hoesler ,
I found one more: https://github.com/apache/airflow/blob/109789283dfda76649fafb7535fee82300e95d28/airflow/models/dagrun.py#L309-L310

`.total_seconds() * 1000` can be removed to avoid unnecessary back & forth conversion.

May you address this together in this PR? After that I can merge immediately.

All other related lines in the code base seems ok.

Cheers!

**Comment by hoesler** [comment]
Sure, I will add that

**Comment by hoesler** [comment]
found one more

**Review by XD-DENG — CHANGES_REQUESTED** [review]
Thanks for the follow-ups @hoesler .
The CI is failing though after the further changes. Let's figure it out first.

File: airflow/models/dagbag.py
Comment (XD-DENG) [inline airflow/models/dagbag.py:423]: One of the error I found in the CI log is below

```
1) ERROR: test_cli_list_dags (tests.cli.commands.test_dag_command.TestCliDags)
----------------------------------------------------------------------
   Traceback (most recent call last):
    tests/cli/commands/test_dag_command.py line 361 in test_cli_list_dags
      dag_command.dag_list_dags(args)
    airflow/utils/cli.py line 80 in wrapper
      return f(*args, **kwargs)
    airflow/cli/commands/dag_command.py line 231 in dag_list_dags
      print(dagbag.dagbag_report())
    airflow/models/dagbag.py line 480 in dagbag_report
      duration=sum([o.duration for o in stats]),
   TypeError: unsupported operand type(s) for +: 'int' and 'datetime.timedelta'
```

It should be due to this change in ` airflow/models/dagbag.py`

Comment (XD-DENG) [reply airflow/models/dagbag.py:423]: For this specific file/change, in order to avoid breaking other logics in other files, I would like to suggest to change in `Stats.timing()` line instead.

I.e., revert your change in line 422, then change

```python
filename = file_stat.file.split('/')[-1].replace('.py', '')
Stats.timing('dag.loading-duration.{}'.
             format(filename),
             format(filename),
             file_stat.duration)
```

into

```python
filename = file_stat.file.split('/')[-1].replace('.py', '')
Stats.timing('dag.loading-duration.{}'.
             format(filename),
             format(filename),
             timedelta(seconds=file_stat.duration))
```
(`timedelta` needs to be imported)


Let me know your opinion? Cheers

**Review by XD-DENG — CHANGES_REQUESTED** [review]

**Comment by hoesler** [comment]
The problem was the sum function, which is basically a reduce(array, +, 0). I had to change the initial value to timedelta()

**Comment by hoesler** [comment]
All tests passed, except `test_on_kill` using sqlite and python 3.6. No idea how my code changes should be involved in this. Maybe a randomly race condition, as this test heavily depends on correct execution order.

**Comment by potiuk** [comment]
`test_on_kill` is known to be flaky.  In the upcoming pytest change it's fixed to be less flaky :).

**Comment by XD-DENG** [comment]
Cool, thanks @hoesler .

To be safe, I restarted the failing test. Will merge once it turns green.

**Review by XD-DENG — APPROVED** [review]

**Comment by XD-DENG** [comment]
Hi @hoesler , please note I updated the subject of your JIRA ticket to better describe the code change here, also for clearer change log.

**Pull request merged**
